modules/stochastic_tools/include/utils/GaussianProcessHandler.h

```diff
@@ -52,11 +52,15 @@ class GaussianProcessHandler
    * @param tao_options Additional options if TAO is used for parameter optimization.
    * @param show_tao Switch to show details of TAO optimization.
    */
-  void setupCovarianceMatrix(const RealEigenMatrix & training_params,
-                             const RealEigenMatrix & training_data,
-                             MooseEnum opt_type,
-                             std::string tao_options = "",
-                             bool show_tao = false);
+  void
+  setupCovarianceMatrix(RealEigenMatrix training_params, // const RealEigenMatrix & training_params
+                        RealEigenMatrix training_data,
+                        MooseEnum opt_type,
+                        std::string tao_options = "",
+                        bool show_tao = false,
+                        const unsigned int & iter_ADAM = 1000,
```

Comment:
I think passing a reference to an unsigned int is more expensive than passing an unsigned int.
```suggestion
                       unsigned int iter_ADAM = 1000,
```